Add enrich_li_class for nested tasklist

primercuervo opened the pull request:
This intends to generate tasklist the same way vimwiki does
for .wiki extensions. Allows for a visualization of
progress in nested tasklists, and customization of
the same via html templates

patrickdavey commented:
Nice work! Especially if you're new to ruby :)

Decisions decisions eh. I reckon, as `vimwiki` does support the fancy state-based lists (even in markdown format), it would be a shame not to support them.

Also, we _don't currently propertly support task lists_ we're in the nice position of just being able to make it work how we want to.

So, I reckon yip, lets go with #29 . If you're happy to tidy up the `rubocop` errors that'd be nice, but honestly - I'm happy to merge it as A) I only added rubocop recently as it's a bit of a mess anyway, and B) this is great work and thanks for spending time to make a decent PR with tests etc!

I did have a little play, and slightly tweaked the `enrich` to:

```ruby
  def enrich_li_class!
    syms_hash = {" ]" => 0, ".]" => 1, "o]" =>  2, "O]" => 3, "X]" => 4}
    checkbox =  /<li>\s*\[[\s.oOX]\]/
    checkbox_start =  /<li>\s*\[/
    @result.gsub!(checkbox) do |m|
      m.sub(checkbox_start, '<li class="done')
       .sub(/[\s.oOX\]]*$/, syms_hash) << '">'
    end
    @result
  end
```

Again, don't really mind if you want to keep it as is. I slightly preferred the above as it's using the same regex `\s` instead of just looking for a single space. I think we should be consistent between the `checkbox` regex and whatever you're doing with the two subs.

Great work though :)  Happy to merge, happy to wait if you want to try fix the code violations.

Thanks!

primercuervo commented:
Your version looks definitely tidier. Thanks for the suggestion! I added it and updated the rubocop checks as well. 

I gotta admit that I like this approach better just because I enjoy seeing those visuals in the HTML.

patrickdavey commented:
I'll update the README and push out a new version in the next hour or so hopefully (assuming my daughter stays alseep ;)

Thanks for the PR & discussion.
